_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme

```diff
@@ -794,6 +850,57 @@ function redhat_www_ddo_default($node) {
   return $ddo;
 }
 
+/**
+ * Gets all taxonomy terms, as labels, referenced by a $node.
+ *
+ * @param Drupal\node\NodeInterface $node
+ *   A node to retrieve taxonomy terms from.
+ *
+ * @return array
+ *   An array of taxonomy terms, keyed by fieldname.
+ */
+function _get_all_taxonomy_terms(NodeInterface $node) {
+  // Attempt to retrieve the taxonomy terms reference by the following fields.
+  $taxonomy_fields = [
+    'field_tax_audience_segment' => 'field_tax_audience_segment',
+    'field_tax_business_unit' => 'field_tax_business_unit',
+    'field_tax_campaign' => 'field_tax_campaign',
+    'field_tax_lifecycle' => 'field_tax_lifecycle',
+    'field_tax_product' => 'field_tax_product',
+    'field_tax_product_line' => 'field_tax_product_line',
+    'field_tax_project' => 'field_tax_project',
+    'field_tax_promotion' => 'field_tax_promotion',
+    'field_tax_region' => 'field_tax_region',
+  ];
+  // Initialize an array to collect taxonomy terms for the Adobe DDO object.
+  $ddo_taxonomies = [];
+
+  foreach ($taxonomy_fields as $taxonomy_field) {
+    // If this $node does not have one of the fields, set an empty array.
+    if (!$node->hasField($taxonomy_field)) {
+      $ddo_taxonomies[$taxonomy_field] = [];
+    }
+    else {
+      // Get the referenced taxonomy terms.
+      $terms = $node->get($taxonomy_field)->referencedEntities();
+
+      // If this $node is referencing taxonomy terms, push them onto the
+      // $ddo_taxonomies array, keyed by the fieldname.
+      if (!empty($terms)) {
+        foreach ($terms as $term) {
+          $ddo_taxonomies[$taxonomy_field][] = $term->id();
```

Comment:
Can we delete a taxonomy term from the vocabulary;  View a node that has the orphaned ID reference and see if this line 891 will throw a PHP fatal error (cannot call method id()).   If someone _could_ delete a taxonomy term,  the orphaned references are not cleaned up immediately, it could break the page.

Reply:
It might just be too late, but I just tried to replicate these steps locally, and not only did everything work without throwing any errors, but I also can't find the field value in my database. I will have to pick this back up in the morning.